src/controllers/connectionManager.ts

```diff
@@ -421,6 +426,77 @@ export default class ConnectionManager {
         };
     }
 
+    public handleNonTSqlNotification(): NotificationHandler<LanguageServiceContracts.NonTSqlParams> {
+        // Using a lambda here to perform variable capture on the 'this' reference
+
+        return async (
+            event: LanguageServiceContracts.NonTSqlParams,
+        ): Promise<void> => {
+            const autoDisable: boolean | undefined = await this._vscodeWrapper
+                .getConfiguration()
+                .get(Constants.configAutoDisableNonTSqlLanguageService);
+            if (autoDisable === false) {
+                return;
+            } else if (autoDisable) {
+                changeLanguageServiceForFile(
+                    SqlToolsServerClient.instance,
+                    event.ownerUri,
+                    Constants.noneProviderName,
+                    this._statusView,
+                );
+            } else {
```

Comment:
some comments helping readers through this flow would be nice

// autoDisable set to false, so do nothing

// autoDisable set to true, so change language service for the file to NONE

// autoDisable setting not set, so ask the user if they want to use it